sklearn/partial_dependence.py

```diff
@@ -0,0 +1,648 @@
+
+def partial_dependence(est, target_variables, grid=None, X=None, output=None,
+                       percentiles=(0.05, 0.95), grid_resolution=100,
+                       method=None):
+    """Partial dependence of ``target_variables``.
+
+    Partial dependence plots show the dependence between the joint values
+    of the ``target_variables`` and the function represented
+    by the ``est``.
+
+    Read more in the :ref:`User Guide <partial_dependence>`.
+
+    Parameters
+    ----------
+    est : BaseEstimator
+        A fitted classification or regression model.
+    target_variables : array-like, dtype=int
+        The target features for which the partial dependency should be
+        computed (size should be smaller than 3 for visual renderings).
+    grid : array-like, shape=(n_points, len(target_variables))
+        The grid of ``target_variables`` values for which the
+        partial dependency should be evaluated (either ``grid`` or ``X``
+        must be specified).
+    X : array-like, shape=(n_samples, n_features)
+        The data on which ``est`` was trained. It is used to generate
+        a ``grid`` for the ``target_variables``. The ``grid`` comprises
+        ``grid_resolution`` equally spaced points between the two
+        ``percentiles``.
+    output : int, optional (default=None)
+        The output index to use for multi-output estimators.
+    percentiles : (low, high), default=(0.05, 0.95)
+        The lower and upper percentile used to create the extreme values
+        for the ``grid``. Only if ``X`` is not None.
+    grid_resolution : int, default=100
+        The number of equally spaced points on the ``grid``.
+    method : {'recursion', 'exact', 'estimated', None}, optional (default=None)
+        The method to use to calculate the partial dependence function:
+
+        - If 'recursion', the underlying trees of ``est`` will be recursed to
+          calculate the function. Only supported for BaseGradientBoosting and
+          ForestRegressor.
+        - If 'exact', the function will be calculated by calling the
+          ``predict_proba`` method of ``est`` for classification or ``predict``
+          for regression on ``X``for every point in the grid. To speed up this
+          method, you can use a subset of ``X`` or a more coarse grid.
+        - If 'estimated', the function will be calculated by calling the
+          ``predict_proba`` method of ``est`` for classification or ``predict``
+          for regression on the mean of ``X``.
+        - If None, then 'recursion' will be used if ``est`` is
+          BaseGradientBoosting or ForestRegressor, and 'exact' used for other
+          estimators.
+
+    Returns
+    -------
+    pdp : array, shape=(n_classes, n_points)
+        The partial dependence function evaluated on the ``grid``.
+        For regression and binary classification ``n_classes==1``.
+    axes : seq of ndarray or None
+        The axes with which the grid has been created or None if
```

Comment:
Thanks for this update! 

I am really happy to put my hands on this and try it with real data, but... what is the best way of trying this new code of PDP for RF in an existing sklearn installation? I replaced the old `sklearn.ensemble.partial_dependence` file by the new one developed by @trevorstephens, but when I run my program there are some importing errors:
```
Traceback (most recent call last):
  File "M:/Documents/workspace/........./pdp.py", line 2, in <module>
    from sklearn.ensemble import RandomForestClassifier
  File "C:\Python34\lib\site-packages\sklearn\ensemble\__init__.py", line 25, in <module>
    from . import partial_dependence
  File "C:\Python34\lib\site-packages\sklearn\ensemble\partial_dependence.py", line 13, in <module>
    from .utils.extmath import cartesian
ImportError: No module named 'sklearn.ensemble.utils'
```
I am aware that this "brute force" replacement is not the best way to go, but I am a really occasional git user, so I prefer not to crush anything :-) Should I replace the entire `sklearn.ensemble` folder by the new one instead?